x-pack/plugins/alerting/server/config.ts

```diff
@@ -8,6 +8,21 @@
 import { schema, TypeOf } from '@kbn/config-schema';
 import { validateDurationSchema } from './lib';
 
+const ruleTypeSchema = schema.object({
+  id: schema.string(),
+  timeout: schema.maybe(schema.string({ validate: validateDurationSchema })),
+});
+
+const rulesSchema = schema.object({
+  execution: schema.object({
+    timeout: schema.maybe(schema.string({ validate: validateDurationSchema })),
```

Comment:
Indeed, it's not used yet.
I had already implemented As the consequence of the misunderstanding we discussed recently (override actions.max by rule type/by connector type) ruleTypeOverrides key was already in the config formatter and i didn't remove it.

 It will be used by: https://github.com/elastic/kibana/issues/124807
 and its implementation is ready and waiting for this PR to be merged :)

Reply:
Ok, sounds good. Given the misunderstanding, this is ok. Let's make sure moving forward these changes stick with the relative PRs.